### PR TITLE
Added Signers in BlockDetails & Updated LatestBlocks and TransactionDetails

### LINKED ISSUE #44: Update block details so that it also show which validators have signed
The block schema has been changed to support showing validator signed status

This is an example
```graphql
query block {
  block(number: 100){
    hash
    number
    signers{
      exist
      signer
      validator{
        name
        score
        validatorGroup{
          name
          commission
        }
      }
    }
  }
}
```

And the output
```json
{
  "data": {
    "block": {
      "hash": "0xa1880391ff4de20862e2673dac4551201d720211b41c84e918f12d34585815e0",
      "number": 100,
      "signers": [
        {
          "exist": false,
          "signer": "0x00D23a06830D50e1da5d3c51C172293BBB52f126",
          "validator": {
            "name": "🔱Masternode24",
            "score": 0.9999314403841,
            "validatorGroup": {
              "name": "Masternode24.de",
              "commission": 0.5
            }
          }
        },
        {
          "exist": true,
          "signer": "0x0223E40d1f93A6Fe5BEf63605992aDA10740E13B",
          "validator": {
            "name": "",
            "score": 0,
            "validatorGroup": {
              "name": "HashQuark",
              "commission": 0.5
            }
          }
        },
        {
          "exist": false,
          "signer": "0x02df14C7F5816aB7A27331dB5D8D3fe6E5a6f03f",
          "validator": {
            "name": "celvaly-0",
            "score": 0.9997003093231,
            "validatorGroup": {
              "name": "celvaly",
              "commission": 0.1
            }
          }
        },
        {
          "exist": true,
          "signer": "0x0A6641d4736767D1f2bCf2450200eF384391B441",
          "validator": {
            "name": "Tessellated Geometry",
            "score": 0.9993556121338,
            "validatorGroup": {
              "name": "Tessellated Geometry Alliance",
              "commission": 0.1
            }
          }
        },
        {
          "exist": false,
          "signer": "0x0b2a9bC0aFb37AB527E910732D091ec3725c0c40",
          "validator": {
            "name": "swiftstaking",
            "score": 0.9999181949608,
            "validatorGroup": {
              "name": "swiftstaking",
              "commission": 0
            }
          }
        },
        {
          "exist": true,
          "signer": "0x173c75C8F1BE201cE89cF426Fe12c9997D709626",
          "validator": {
            "name": "AGx1",
            "score": 0.9999314403841,
            "validatorGroup": {
              "name": "AG",
              "commission": 0.2
            }
          }
        },
        {
          "exist": true,
          "signer": "0x198958f0b860AB0E3937F468FE366AAc9EEbaD2e",
          "validator": {
            "name": "sparkpool-v1",
            "score": 0.9487285352794,
            "validatorGroup": {
              "name": "sparkpool-group",
              "commission": 0.1
            }
          }
        },
        {
          "exist": false,
          "signer": "0x273faE03D02B80477c574542d989C14950A3eA24",
          "validator": {
            "name": "Alive29x1",
            "score": 0.9922766736961,
            "validatorGroup": {
              "name": "chainvibes",
              "commission": 0.5
            }
          }
        },
        {
          "exist": false,
          "signer": "0x2E0a1C2048B41522a833A2FA4FA919A0e8B9b1d3",
          "validator": {
            "name": "wotrust1",
            "score": 0.9999314403841,
            "validatorGroup": {
              "name": "wotrust",
              "commission": 1
            }
          }
        },
        {
          "exist": true,
          "signer": "0x2Eb79345089cA6F703F3b3C4235315CbeAaD6D3C",
          "validator": {
            "name": "nonce",
            "score": 0.9933160527442,
            "validatorGroup": {
              "name": "nonce",
              "commission": 0.1
            }
          }
        },
        {
          "exist": true,
          "signer": "0x2a1BfF2452AcA1ca5FFABbD34b2744109d11E4f5",
          "validator": {
            "name": "huglester",
            "score": 0.9978808306474,
            "validatorGroup": {
              "name": "huglester",
              "commission": 0.1
            }
          }
        },
        {
          "exist": false,
          "signer": "0x2aD6D354e3aB2A196B64964a788534fdF7932e2B",
          "validator": {
            "name": "Cypher Core",
            "score": 0.9988678856422,
            "validatorGroup": {
              "name": "Cypher Core",
              "commission": 0.1
            }
          }
        },
        {
          "exist": false,
          "signer": "0x3ed95D6D4Ce36Ea7B349cD401e324316D956331a",
          "validator": {
            "name": "",
            "score": 0.9998637865309,
            "validatorGroup": {
              "name": "",
              "commission": 0.5
            }
          }
        },
        {
          "exist": true,
          "signer": "0x436d12F639A32509685080161Ff0365Fc15545f0",
          "validator": {
            "name": "🌴👦",
            "score": 0.7458134171671,
            "validatorGroup": {
              "name": "❄️❄️❄️ + 🌴👦 + Danil Ushakov = ✅🚀🌟",
              "commission": 0.3
            }
          }
        },
        {
          "exist": true,
          "signer": "0x43882141555003B3e71110f567373b59Ac4cb0BD",
          "validator": {
            "name": "",
            "score": 0.9958716235495,
            "validatorGroup": {
              "name": "",
              "commission": 0.5
            }
          }
        },
        {
          "exist": true,
          "signer": "0x464CCE7999B3D90a8F1fFaf94A71FBDeA1E65435",
          "validator": {
            "name": "BitCat",
            "score": 0.8332281830026,
            "validatorGroup": {
              "name": "BitCat",
              "commission": 0.1
            }
          }
        },
        {
          "exist": false,
          "signer": "0x482AD8050622483a16aeea36272A4D3eA000C018",
          "validator": {
            "name": "Chorus One Validator1",
            "score": 0.982532028619,
            "validatorGroup": {
              "name": "Chorus One",
              "commission": 0.075
            }
          }
        },
        {
          "exist": true,
          "signer": "0x4A03C4c2E101AC4612d89b79f61c9C5BDd51929D",
          "validator": {
            "name": "chainvibes",
            "score": 0.9999031493719,
            "validatorGroup": {
              "name": "chainvibes",
              "commission": 0.5
            }
          }
        },
        {
          "exist": true,
          "signer": "0x4CB90Ebba92141eD3021F5dC4e6C8bb642095846",
          "validator": {
            "name": "dynamic",
            "score": 0.9980182277173,
            "validatorGroup": {
              "name": "dynamic",
              "commission": 0.1
            }
          }
        },
        {
          "exist": false,
          "signer": "0x503ae8e26403BD8D41C21f73E75A3C118D2c0743",
          "validator": {
            "name": "Bi23",
            "score": 0.9903230029306,
            "validatorGroup": {
              "name": "Bi23 Labs",
              "commission": 0.1
            }
          }
        },
        {
          "exist": false,
          "signer": "0x591cB5E536aa19Fe5f26593739FE8ce717142e59",
          "validator": {
            "name": "Pretoria Validator A",
            "score": 0.9999314403841,
            "validatorGroup": {
              "name": "PretoriaResearchLab.io",
              "commission": 0.1
            }
          }
        },
        {
          "exist": true,
          "signer": "0x59F7b67e6BeAE0223DDC91Eec010b670c553E8e0",
          "validator": {
            "name": "",
            "score": 0.9993292928542,
            "validatorGroup": {
              "name": "Chainflow",
              "commission": 0.7
            }
          }
        },
        {
          "exist": true,
          "signer": "0x5b55452Bdba5971D606F47647bd383f3C3fa7285",
          "validator": {
            "name": "Nodeasy",
            "score": 0.7700976544609,
            "validatorGroup": {
              "name": "www.nodeasy.com",
              "commission": 0.5
            }
          }
        },
        {
          "exist": false,
          "signer": "0x613F809e82F7BcbbFB86f939684ac080454400fA",
          "validator": {
            "name": "TDlabs",
            "score": 0.9468775995301,
            "validatorGroup": {
              "name": "TDlabs | Staked.us",
              "commission": 0.1
            }
          }
        },
        {
          "exist": true,
          "signer": "0x63b4b616c5345e3DcC9e21dB69297e2129447f4e",
          "validator": {
            "name": "syncnode",
            "score": 0.9990808830487,
            "validatorGroup": {
              "name": "syncnode_group",
              "commission": 0
            }
          }
        },
        {
          "exist": true,
          "signer": "0x69901924E6c045A03CC163c6b8AE8AF80fA1ee80",
          "validator": {
            "name": "YYYYYYYYYYYYY1",
            "score": 0.9999314403841,
            "validatorGroup": {
              "name": "YYYYYYYYYYYYY",
              "commission": 0.5
            }
          }
        },
        {
          "exist": true,
          "signer": "0x74288dbEFa3a55986c039953b67139A466474fC4",
          "validator": {
            "name": "Projecttent 1",
            "score": 0.9567276635953,
            "validatorGroup": {
              "name": "Projecttent",
              "commission": 0.1
            }
          }
        },
        {
          "exist": true,
          "signer": "0x82f0E7879314516952f7961b15C63FC6B2734DFB",
          "validator": {
            "name": "kytzu",
            "score": 0.9990351784297,
            "validatorGroup": {
              "name": "Kytzu Group",
              "commission": 0.1
            }
          }
        },
        {
          "exist": false,
          "signer": "0x88cB5A11a6e451ede2376457E718EBBcD0799c14",
          "validator": {
            "name": "censusworks.0",
            "score": 0.9999314403841,
            "validatorGroup": {
              "name": "Censusworks",
              "commission": 1
            }
          }
        },
        {
          "exist": false,
          "signer": "0x953f2501B18a3FdCF959aA901e74D5308e4D522f",
          "validator": {
            "name": "SlavaMo",
            "score": 0.9999314403841,
            "validatorGroup": {
              "name": "SlavaMo",
              "commission": 0.999
            }
          }
        },
        {
          "exist": false,
          "signer": "0xA3ba418DBd6e2F2a0eDeBAe256F988FCC92b8E86",
          "validator": {
            "name": "Staking Fund #1",
            "score": 0.9988264896841,
            "validatorGroup": {
              "name": "Staking Fund",
              "commission": 0
            }
          }
        },
        {
          "exist": false,
          "signer": "0xA90640bF05711e5674daCF5A4368Aa6348031e01",
          "validator": {
            "name": "stakevalley01",
            "score": 0.9999314403841,
            "validatorGroup": {
              "name": "stakevalley",
              "commission": 0.99
            }
          }
        },
        {
          "exist": false,
          "signer": "0xAf35c24f9644e3743FFCba9C17980492C31D75a4",
          "validator": {
            "name": "gunray",
            "score": 0.9999314403841,
            "validatorGroup": {
              "name": "gunray",
              "commission": 0.1
            }
          }
        },
        {
          "exist": true,
          "signer": "0xBc6963fc0e2f5547bA949ed39e80b8388321104f",
          "validator": {
            "name": "",
            "score": 0.9973607777263,
            "validatorGroup": {
              "name": "",
              "commission": 0.5
            }
          }
        },
        {
          "exist": true,
          "signer": "0xC46Dc0741FF61af883E284daEA062ae7382E7091",
          "validator": {
            "name": "Easy 2 Stake",
            "score": 0.9965578410595,
            "validatorGroup": {
              "name": "Easy2Stake Group",
              "commission": 0.1
            }
          }
        },
        {
          "exist": false,
          "signer": "0xCBc1013EE337FB25a600daF2f73Bf270322F5B1e",
          "validator": {
            "name": "moonli.me",
            "score": 0.9880415863876,
            "validatorGroup": {
              "name": "moonli.me",
              "commission": 0
            }
          }
        },
        {
          "exist": false,
          "signer": "0xD375E40327b48738234f385B98056cEFc6CaD73a",
          "validator": {
            "name": "The Passive Trust - V1",
            "score": 0.9995782009897,
            "validatorGroup": {
              "name": "The Passive Trust",
              "commission": 0.4
            }
          }
        },
        {
          "exist": false,
          "signer": "0xD6d1811894bf2E0FF2b372eCfe084Be08EbF7De2",
          "validator": {
            "name": "nga_validator",
            "score": 0.9999254793007,
            "validatorGroup": {
              "name": "nga_group",
              "commission": 0.5
            }
          }
        },
        {
          "exist": false,
          "signer": "0xE87F2D4B87540D51e22a597bF747CEef8F32b29F",
          "validator": {
            "name": "blockdaemon-0",
            "score": 0.9641284528204,
            "validatorGroup": {
              "name": "Blockdaemon",
              "commission": 0.1
            }
          }
        },
        {
          "exist": false,
          "signer": "0xF027A6b28bDcA9B9F95e7158a9D3a46BA7f4589E",
          "validator": {
            "name": "NewroadNetwork",
            "score": 0.9996819240457,
            "validatorGroup": {
              "name": "The Celo Group",
              "commission": 0.08
            }
          }
        },
        {
          "exist": false,
          "signer": "0xF139e74aDEC329E715AB49a68c5548A00e40CBc2",
          "validator": {
            "name": "warfollowsme | Celomap.io",
            "score": 0.8137812363904,
            "validatorGroup": {
              "name": "Ryabina | Celomap.io",
              "commission": 0.1
            }
          }
        },
        {
          "exist": true,
          "signer": "0xFFbCF262C1d5c4392ef469BA79f2CD195d2AffDa",
          "validator": {
            "name": "Simply Staking",
            "score": 0.9993350055736,
            "validatorGroup": {
              "name": "Simply Staking",
              "commission": 0
            }
          }
        },
        {
          "exist": false,
          "signer": "0xaAf29578F2DbC2Aac1b0D906a326512A8D706335",
          "validator": {
            "name": "StakesStone",
            "score": 0.9916950813931,
            "validatorGroup": {
              "name": "StakesStoneGroup",
              "commission": 0.1
            }
          }
        },
        {
          "exist": false,
          "signer": "0xaeD733bb20921B682EB35bb89BD398F604CcD5BC",
          "validator": {
            "name": "Perfect_Stake.1",
            "score": 0.7830513380413,
            "validatorGroup": {
              "name": "Perfect Stake",
              "commission": 0.1
            }
          }
        },
        {
          "exist": false,
          "signer": "0xc328f76b1Ed2EeAcd4460b11435f4BD48BF2361d",
          "validator": {
            "name": "HappyCelo-val-1",
            "score": 0.9982182504268,
            "validatorGroup": {
              "name": "HappyCelo",
              "commission": 0
            }
          }
        },
        {
          "exist": true,
          "signer": "0xc6f916aD6E360651Bb95f8e67C1C28805745d084",
          "validator": {
            "name": "Validator.Capital1",
            "score": 0.9999210900925,
            "validatorGroup": {
              "name": "Validator.Capital",
              "commission": 0.45
            }
          }
        },
        {
          "exist": false,
          "signer": "0xcF514E382efF83752fBafeC2D836A3efef7fD039",
          "validator": {
            "name": "G&S 1 ",
            "score": 0.9999314403841,
            "validatorGroup": {
              "name": "Goods & Services",
              "commission": 0.1
            }
          }
        },
        {
          "exist": true,
          "signer": "0xd8C68eBeCb6F074ac5C4FB66a690AC0Ad38a5a3c",
          "validator": {
            "name": "DSRV - CeloWhale.com - 1",
            "score": 0.9950398978204,
            "validatorGroup": {
              "name": "DSRV CeloWhale.com",
              "commission": 0.2
            }
          }
        },
        {
          "exist": true,
          "signer": "0xe10A8Cc6c22CDc320c67BD600a1d8a0a46D7f400",
          "validator": {
            "name": "Figment Networks 1",
            "score": 0.9879512819581,
            "validatorGroup": {
              "name": "Figment Networks",
              "commission": 0
            }
          }
        },
        {
          "exist": false,
          "signer": "0xe297b2fFAeC13E2E0CA299Db0D97c1Dcad3d04bF",
          "validator": {
            "name": "Conclave-I",
            "score": 0.9999314403841,
            "validatorGroup": {
              "name": "Conclave",
              "commission": 0.4
            }
          }
        },
        {
          "exist": false,
          "signer": "0xe3b1e1647BECe359C76582550F6E210999973B50",
          "validator": {
            "name": "ChainLayer1",
            "score": 0.9996483115027,
            "validatorGroup": {
              "name": "ChainLayer",
              "commission": 0.1
            }
          }
        },
        {
          "exist": true,
          "signer": "0xe5617da4dFa6CE4912F1D39889FE2dEc2F7E66b3",
          "validator": {
            "name": "MoonletV1",
            "score": 0.9908140181608,
            "validatorGroup": {
              "name": "Moonlet.io",
              "commission": 0.1
            }
          }
        }
      ]
    }
  }
}
```

### PR DESCRIPTION
**Pull request by MonikaCat**
* Added Signers in BlockDetails
* Updated LatestBlocks and TransactionDetails

Fixes #44

### CODE REVIEW

File: components/block/BlockDetails.tsx [hunk 1]
```diff
@@ -153,7 +151,7 @@ const BlockDetails = () => {
             <Divider variant="middle" className={classes.divider} />
           </Grid>
 
-          <Grid item xs={12} className={classes.item}>
+          <Grid item xs={12} >
             <Typography variant="body2" component="h2">
               Miner
```
Comment (kwunyeung) [inline components/block/BlockDetails.tsx:156]:
```suggestion
              Validator
```

File: components/block/BlockDetails.tsx [hunk 2]
```diff
@@ -251,6 +249,33 @@ const BlockDetails = () => {
...
+            <Typography variant="body2" component="h2" gutterBottom>
+              Signers
+            </Typography>
+            {data.block && data.block.signers
+              ?
+              data.block.signers.map((row: any, index: number) => {
+                return (
+                  <span key={index}>
+
+                    <Typography variant="body2" gutterBottom >
```
Comment (kwunyeung) [inline components/block/BlockDetails.tsx:263]: We better show this as a list indicating if the validator has signed the block. The current implementation doesn't show whether a validator is missing in this block.

Two approaches:
1. List as a table with the first column the validator, second column whether it's up or down
2. Two tabs list out all up validators and down validators.

File: components/block/BlockDetails.tsx [hunk 3]
```diff
@@ -251,6 +249,33 @@ const BlockDetails = () => {
...
+                      <Link
+                        href="/validatorGroup/[validatorGroupDetails]/"
+                        as={row.validator && row.validator.validatorGroup && row.validator.validatorGroup.address ?
+                          `/validatorGroup/${row.validator.validatorGroup.address}` : ""}
```
Comment (kwunyeung) [inline components/block/BlockDetails.tsx:267]: I wonder if we should have an individual page for each validator

File: components/block/BlockDetails.tsx (hunk 2 above)
Comment (MonikaCat) [reply components/block/BlockDetails.tsx]: Ok! I have updated the code and added Tab List with Signed & Missed Blocks

File: components/block/BlockDetails.tsx (hunk 3 above)
Comment (MonikaCat) [reply components/block/BlockDetails.tsx]: I have created a separate issue for this  - we can track it here https://github.com/forbole/big-dipper-celo-client/issues/62

File: components/block/BlockDetails.tsx [hunk 4]
```diff
@@ -67,13 +113,47 @@ const BlockDetails = () => {
   const { loading, error, data } = useQuery(GET_BLOCK_DETAILS, {
```
Comment (kwunyeung) [inline components/block/BlockDetails.tsx:113]: Get error when rendering this page at this line.
```
Unhandled Runtime Error
Error: Rendered more hooks than during the previous render.
```

File: components/block/LatestBlocks.tsx [hunk 5]
```diff
@@ -316,7 +316,7 @@ const LatestBlocks = ({ pagination, displayCard }: LatestBlocksProps) => {
                             <Link href="#" color="secondary">
                               {row.transactions &&
                                 row.transactions.transactionIndex
-                                ? row.transactions.transactionIndex.length()
+                                ? row.transactions.transactionIndex
```
[the remark this answers is not in the document]
Comment (MonikaCat) [reply components/block/LatestBlocks.tsx]: You're right.. Updated!

File: components/block/BlockDetails.tsx (hunk 4 above)
Comment (MonikaCat) [reply components/block/BlockDetails.tsx:110]: Updated!

**Pull request merged**